### PR TITLE
Feature/user backend

### PR DESCRIPTION
**Pull request by rkuenzi-epfl**
The LoginActivity is now connected with the user backend. Once there is tests for the UI we can merge with main.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -44,6 +44,7 @@ captures/
 .idea/assetWizardSettings.xml
 .idea/dictionaries
 .idea/libraries
+.idea/caches
```
Comment (stblr) [inline .gitignore:47]: Already in the gitignore (two lines below)

File: app/build.gradle [hunk 2]
```diff
@@ -36,15 +36,23 @@ android {
 
 dependencies {
 
+    implementation 'com.google.android.gms:play-services-auth:19.0.0'
```
Comment (stblr) [inline app/build.gradle:39]: Duplicate

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -9,6 +9,7 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/Theme.Wnder">
+	<activity android:name=".LoginActivity"></activity>
```
Comment (stblr) [inline app/src/main/AndroidManifest.xml:12]: Indentation is off

File: app/src/test/java/com/github/wnder/UserTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    @Test
+    public void signedInUserReturnCorrectProfilePictureUri(){
+        // Reusing default Profile Picture for testing
+        User u = new SignedInUser("TestUser", Uri.parse("android.resource://raw/ladiag.jpg"));
+        assertEquals(u.getProfilePicture(), Uri.parse("android.resource://raw/ladiag.jpg"));
+    }
+
+    @Test
+    public void globalUserReturnGuestUserByDefault(){
+
```
Comment (stblr) [inline app/src/test/java/com/github/wnder/UserTest.java:44]: Blank line

File: app/src/main/java/com/github/wnder/SignedInUser.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    private Uri profilePicture;
+    // These are guesses on future fields for a user
+    //private int GlobalScore;
+    //private History history;
+    
+    public SignedInUser(String name, Uri profilePicture){
+        //this.account = account;
+        this.name = name;
+        this.profilePicture = profilePicture;
+
```
Comment (stblr) [inline app/src/main/java/com/github/wnder/SignedInUser.java:21]: Blank line

File: app/src/main/java/com/github/wnder/LoginActivity.java [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+        GoogleSignInOptions gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestEmail()
+                .build();
+
+        client = GoogleSignIn.getClient(this, gso);
+    }
+
+    @Override
+    protected void onStart(){
```
Comment (stblr) [inline app/src/main/java/com/github/wnder/LoginActivity.java:46]: Maybe always put a space between `()` and `{` in function declarations

File: app/src/main/java/com/github/wnder/LoginActivity.java [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.tasks.Task;
+
+public class LoginActivity extends AppCompatActivity implements View.OnClickListener {
+
+    private View signInButton;
+    private TextView textLogin;
+    private GoogleSignInClient client;
+    private int RC_SIGN_IN = 10; //nombre arbitraire
```
Comment (stblr) [inline app/src/main/java/com/github/wnder/LoginActivity.java:24]: Should all comments be written in English?

File: app/src/androidTest/java/com/github/wnder/MainActivityTest.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+
+@RunWith(AndroidJUnit4.class)
+public class MainActivityTest {
+
+    //Open the activity as a test because there is nothing in the activity yet
```
Comment (stblr) [inline app/src/androidTest/java/com/github/wnder/MainActivityTest.java:20]: Consistently put a space after `//`

**Review by stblr — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/wnder/LoginActivity.java (hunk 7 above)
Comment (rkuenzi-epfl) [reply app/src/main/java/com/github/wnder/LoginActivity.java:24]: Yes they should, these were written by @leonardvaney I forgot to remove or translate them

File: app/build.gradle (hunk 2 above)
Comment (rkuenzi-epfl) [reply app/build.gradle:39]: I'll remove it

File: .gitignore (hunk 1 above)
Comment (rkuenzi-epfl) [reply .gitignore:47]: Thanks, git kept doing odd stuff with those files so I thought it wasn't there.

File: app/src/main/java/com/github/wnder/LoginActivity.java [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.tasks.Task;
+
+public class LoginActivity extends AppCompatActivity implements View.OnClickListener {
+
+    private View signInButton;
+    private TextView textLogin;
+    private GoogleSignInClient client;
+    private int RC_SIGN_IN = 10; // Arbitrary number
+    private String TAG = "LoginActivity.java";
```
Comment (JeremyDezalos) [inline app/src/main/java/com/github/wnder/LoginActivity.java:25]: Constants should be final

**Review by leonardvaney — APPROVED** [review]
Some warnings for the hardcoded string in LoginActivity but otherwise it's ok.

**Review by JeremyDezalos — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/wnder/LoginActivity.java (hunk 9 above)
Comment (rkuenzi-epfl) [reply app/src/main/java/com/github/wnder/LoginActivity.java]: Yes, thanks

**Review by JeremyDezalos — APPROVED** [review]

**Review by stblr — APPROVED** [review]

**Pull request merged**